### DISCUSSION TITLE
Auth0 Auth Provider

### DISCUSSION THREAD

**Discussion by joelbqz** [strapi/strapi#7616] [Legacy Topics]
Hello everyone, I added Auth0 as login provider for a project I'm working on, I did by following instructions in the documentation https://strapi.io/documentation/v3.x/plugins/users-permissions.html#adding-a-new-provider-to-your-project

But I would love to re-use this in other projects, so, I was wondering If you can point me in the right direction to make this integration a plugin so it can be easier to maintain.

**Comment by joelbqz** [comment]
PD. I understand this shows how to build a local plugin https://strapi.io/documentation/v3.x/plugin-development/plugin-architecture.html

But I would like to know how to override default plugins using local plugins

**Reply by derrickmehaffy** [reply]
You can use the Node package.json local modules feature which allows you to call a module from the local file system: https://www.stefanjudis.com/today-i-learned/npm-install-supports-local-modules/

**Reply by joelbqz** [reply]
@derrickmehaffy thanks for your reply, maybe I didn't formulate my question well. I did an Auth0 integration which works very well, but this integration was made following Strapi documentation and editing extension files, I would like to make this integration a plugin, so I can publish it and anyone can install and use It without having to make changes in their extension files.

Is there a way I can create a strapi local plugin that can override some files of other plugins?

**Reply by derrickmehaffy** [reply]
Not directly no, and due to how we load some the plugins, publishing an entire replacement won't work directly in most cases (for core plugins). So in this way, you have to forcefully override it using the method I described above, or users will have to use extensions to modify the plugin.

**Reply by joelbqz** [reply]
> Not directly no, and due to how we load some the plugins, publishing an entire replacement won't work directly in most cases (for core plugins). So in this way, you have to forcefully override it using the method I described above, or users will have to use extensions to modify the plugin.

I don't want to provide an entire replacement for any core plugin, I just want to provide a new plugin that overrides 1-2 files of core plugins(user-permissions plugin).

**Reply by joelbqz** [reply]
@derrickmehaffy sorry, let me see If I understand correctly, with the method you provided, I can override just the files I need to on the core plugins?

**Reply by derrickmehaffy** [reply]
The way our plugin system currently functions, I'm not sure if that is possible. Maybe with our new plugin refresh: https://portal.productboard.com/strapi/1-roadmap/c/72-plugin-api-v4 which will do away with our current extensions system to something that would make it a little easier.
